### PR TITLE
feat: adding iam policy, doc: create readme

### PR DESCRIPTION
**Pull request by samen93**
doc: update readme

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,301 @@
...
+
+    Textual representation of an expression in Common Expression Language syntax.
+
+  - **`title`**: **_(Required `string`)_**
+
+    A title for the expression, i.e. a short string describing its purpose.
+
+  - **`description`**: _(Optional `string`)_
+
+    An optional description of the expression. This is a longer text which describes the expression, e.g. when hovered over it in a UI.
```
Comment (mariux) [inline README.md:184]: wrong indent

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (samen93) [reply README.md:184]: i don't see it broken in visual cod
Screenshot from 2021-11-04 12-13-06
e

File: main.tf [hunk 2]
```diff
@@ -1,5 +1,5 @@
-resource "google_project_iam_binding" "project" {
-  count = var.module_enabled && var.authoritative ? 1 : 0
+resource "google_project_iam_binding" "binding" {
```
Comment (mariux) [inline main.tf:1]: this is an incompatible change... please revert resource will be recreated.

**Review by mariux — CHANGES_REQUESTED** [review]
please do not rename resources this will break eisting infrastructure

File: variables.tf [hunk 3]
```diff
@@ -36,6 +36,12 @@ variable "condition" {
   default     = null
 }
 
+variable "policy_bindings" {
+  description = "(Optional) A list of IAM policy bindings."
+  type        = any
+  default     = null
+}
+
```
Comment (mariux) [inline variables.tf:44]: i assume this makes role optional... but it is still required...please check in all modules...

**Review by mariux — CHANGES_REQUESTED** [review]

File: README.md [hunk 4]
```diff
@@ -1 +1,276 @@
...
+- **`members`**: _(Optional `set(string)`)_
+
+  Identities that will be granted the privilege in role. Each entry can have one of the following values:
+  - `user:{emailid}`: An email address that represents a specific Google account. For example, alice@gmail.com or joe@example.com.
+  - `serviceAccount:{emailid}`: An email address that represents a service account. For example, my-other-app@appspot.gserviceaccount.com.
+  - `group:{emailid}`: An email address that represents a Google group. For example, admins@example.com.
+  - `domain:{domain}`: A G Suite domain (primary, instead of alias) name that represents all the users of that domain. For example, google.com or example.com.
+  - `projectOwner:projectid`: Owners of the given project. For example, `projectOwner:my-example-project`
+  - `projectEditor:projectid`: Editors of the given project. For example, `projectEditor:my-example-project`
+  - `projectViewer:projectid`: Viewers of the given project. For example, `projectViewer:my-example-project`
```
Comment (mariux) [inline README.md:99]: where is this documented to work?

**Review by mariux — CHANGES_REQUESTED** [review]

File: variables.tf [hunk 5]
```diff
@@ -8,26 +8,27 @@ variable "project" {
...
+  description = "(Optional) Identities that will be granted the privilege in role. Each entry can have one of the following values: 'user:{emailid}', 'serviceAccount:{emailid}', 'group:{emailid}', 'domain:{domain}'."
   type        = set(string)
   default     = []
 }
 
 variable "authoritative" {
   description = "(Optional) Whether to exclusively set (authoritative mode) or add (non-authoritative/additive mode) members to the role."
   type        = bool
-  default     = false
+  default     = true
```
Comment (mariux) [inline variables.tf:31]: this breaks compatibility..

Comment (mariux) [inline variables.tf:31]:
```suggestion
  default     = false
```

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]

File: README.md [hunk 6]
```diff
@@ -1 +1,273 @@
...
+  - `group:{emailid}`: An email address that represents a Google group. For example, admins@example.com.
+  - `domain:{domain}`: A G Suite domain (primary, instead of alias) name that represents all the users of that domain. For example, google.com or example.com.
+
+  Default is `[]`.
+
+- **`authoritative`**: _(Optional `bool`)_
+
+  Whether to exclusively set (authoritative mode) or add (non-authoritative/additive mode) members to the role.
+
+  Default is `true`.
```
Comment (mariux) [inline README.md:104]:
```suggestion
  Default is `false`.
```

**Review by mariux — CHANGES_REQUESTED** [review]

**Review by mariux — APPROVED** [review]

**Pull request merged**